packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -228,13 +286,86 @@ We may also implement an external queue canceling functionality.
+
+The current proposal is for the highlight table to be used as a filter for the main dies table. This can be realized by using the [`semijoin`](https://uwdata.github.io/arquero/api/verbs#semijoin) operation from the Arquero library. this will function as follows.
+
+The main table:
+
+| (index) | colIndex | rowIndex | value              | firstTag | secondTag |
+| ------- | -------- | -------- | ------------------ | -------- | --------- |
+| 0       | 0        | 2        | 14.239999771118164 | a        | b         |
+| 1       | 1        | 2        | 76.43000030517578  | b        | c         |
+| 2       | 1        | 1        | 44.630001068115234 | g        | null      |
+| 3       | 1        | 3        | 67.93000030517578  | a        | null      |
+| 4       | 2        | 2        | 72.70999908447266  | h        | e         |
+| 5       | 2        | 1        | 79.04000091552734  | b        | null      |
+| 6       | 2        | 0        | 26.489999771118164 | c        | null      |
+| 7       | 2        | 3        | 37.790000915527344 | null     | null      |
+| 8       | 2        | 4        | 59.81999969482422  | null     | null      |
+| 9       | 3        | 2        | 52.900001525878906 | null     | null      |
+| 10      | 3        | 1        | 98.5               | g        | null      |
+| 11      | 3        | 3        | 20.829999923706055 | c        | null      |
+| 12      | 4        | 2        | 62.79999923706055  | g        | null      |
+
+The highlight table:
+
+| (index) | firstTag |
+| ------- | -------- |
+| 0       | a        |
+| 1       | b        |
+| 2       | c        |
+
+The filtered table:
+
+| (index) | colIndex | rowIndex | value              | firstTag | secondTag |
+| ------- | -------- | -------- | ------------------ | -------- | --------- |
+| 0       | 0        | 2        | 14.239999771118164 | a        | b         |
+| 1       | 1        | 2        | 76.43000030517578  | b        | c         |
+| 2       | 1        | 3        | 67.93000030517578  | a        | null      |
+| 3       | 2        | 1        | 79.04000091552734  | b        | null      |
+| 4       | 2        | 0        | 26.489999771118164 | c        | null      |
+| 5       | 3        | 3        | 20.829999923706055 | c        | null      |
+
+The filter matched the rows with the same values from the highlight table. This can be used for tags when filtering value ranges, values themselves, column and row indexes or other types of supported data types.
+
+The details of the implementation and more refined filtering will be discussed.
+
+Anther option is using the existing `highlightedTags` API with a `List` column in the table [(listed as supported)](https://arrow.apache.org/docs/status.html).
+
```

Comment:
```suggestion
Specific open questions:
- For a highlightedTags table api:
  - Should the API be constrained in the [supported arrow types](https://arrow.apache.org/docs/status.html)?
    - Should just primitive types be supported?
    - Should just types supported in the JavaScript and C# languages be supported?
    - Should just types well-supported in all exisiting implementation be supported?
    - Would certain types, ex. like strings, lead to poor performance and be discouraged?
    - Are there implementation challenges transferring buffers of arbitrary types across the Web Worker boundary?
  - Should the API be constrained in the set of columns that participate in highlighting instead of all columns? Maybe columns with a specific name prefix like `highlighted_`?
  - Is there real known benefits for specifying per floating point `value`? Or specific `rowIndex` / `columnIndex` independently?
  - Does the highlightTable need to contain all columns used in diesTable? Can it just be a subset of columns?
- For a tags columns api:
  - Do columns of List<string> work in tables?
  - Do dictionary columns work in tables to improve efficiency compared to Liist<string>?
  - What is the performance of a List<string> / Dictionary column api compared to the alternatives?
```